src/auth/strategies/local.strategy.ts

```diff
@@ -15,7 +15,7 @@ export class LocalStrategy extends PassportStrategy(Strategy) {
     const user = await this.authService.validateUser(data);
 
     if (!user) {
-      throw new UnauthorizedException();
+      throw new NotFoundException();
```

Comment:
I think the `UnauthorizedException()` is an appropriate error here. Now we are explicitly telling user that "hey, this user does not exist in the system", thus enabling malicious actors to perform a user enumeration attack. Note that `this.authService.validateUser(data);` method can return `UnauthorizedException()` on its own. We should return `UnauthorizedException()` here

Reply:
Ok I get it. If so then should I also change the Exception within `signIn` method in `auth.service`?

Reply:
if the logic is exactly the same as above, then yes, I would change it 👍🏻